packages/mobile/src/transactions/feed/TransactionFeed.test.tsx

```diff
@@ -0,0 +1,126 @@
+import { render, waitFor } from '@testing-library/react-native'
+import { FetchMock } from 'jest-fetch-mock/types'
+import React from 'react'
+import { Provider } from 'react-redux'
+import { RootState } from 'src/redux/reducers'
+import TransactionFeed from 'src/transactions/feed/TransactionFeed'
+import { TokenTransaction, TokenTransactionTypeV2 } from 'src/transactions/types'
+import { createMockStore, RecursivePartial } from 'test/utils'
+import { mockCusdAddress } from 'test/values'
+
+const MOCK_RESPONSE: { data: { tokenTransactionsV2: { transactions: TokenTransaction[] } } } = {
+  data: {
+    tokenTransactionsV2: {
+      transactions: [
+        {
+          __typename: 'TokenTransferV2',
+          address: '0xd68360cce1f1ff696d898f58f03e0f1252f2ea33',
+          amount: {
+            tokenAddress: mockCusdAddress,
+            value: '0.1',
+          },
+          block: '8648978',
+          fees: [],
+          metadata: {},
+          timestamp: 1638600563000,
+          transactionHash: '0x544367eaf2b01622dd1c7b75a6b19bf278d72127aecfb2e5106424c40c268e8b',
+          type: TokenTransactionTypeV2.Received,
+        },
+        {
+          __typename: 'TokenTransferV2',
+          address: '0xd68360cce1f1ff696d898f58f03e0f1252f2ea33',
+          amount: {
+            tokenAddress: '0xrandomAddress',
+            value: '0.1',
+          },
+          block: '8648978',
+          fees: [],
+          metadata: {},
+          timestamp: 1638600563000,
+          transactionHash: '0x544367eaf2b01622dd1c7b75a6b19bf278d72127aecfb2e5106424c40c268e8b',
+          type: TokenTransactionTypeV2.Received,
+        },
+      ],
+    },
+  },
+}
+
+describe('TransactionFeed', () => {
+  const mockFetch = fetch as FetchMock
+  beforeEach(() => {
+    jest.useRealTimers()
+    jest.clearAllMocks()
+    mockFetch.resetMocks()
+  })
+
+  function renderScreen(storeOverrides: RecursivePartial<RootState> = {}) {
+    const store = createMockStore({
+      ...storeOverrides,
+    })
+
+    const tree = render(
+      <Provider store={store}>
+        <TransactionFeed />
+      </Provider>
+    )
+
+    return {
+      store,
+      ...tree,
+    }
+  }
+
+  it('renders correctly when there is a response', async () => {
+    mockFetch.mockResponse(JSON.stringify(MOCK_RESPONSE))
+
+    const tree = renderScreen({})
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+
+    expect(tree.queryByTestId('NoActivity/loading')).toBeNull()
+    expect(tree.queryByTestId('NoActivity/error')).toBeNull()
+
+    expect(tree).toMatchSnapshot()
```

Comment:
This is probably a wider discussion but thought this might be a good conversation starter - do we want to keep using snapshot tests? My personal opinion is that they capture mostly implementation detail, and while sometimes during a refactor it can be a nice prompt that this component was affected, most of the time we will update sanpshot tests without a second thought and the snapshot updates are not reviewed. The end result is that they add extra maintainence overhead and don't really give us any more confidence that the component is working correctly...they also do a less good job of acting as documentation (it's easy to see that in this scenario we should not see the loading or error states in the first 2 `expect` statements) 

so TL;DR I think that the rest of the tests are great and this snapshot isn't really needed - what do you think?

Reply:
I agree in general, snapshot files don't seem very useful. I'm not sure about starting in this PR though, maybe this is something we can talk over together in our next team meeting? So that we can all agree and move forward in agreement :)

Reply:
yeah of course! leave it for now and i'll bring it up for the next meeting :)